### PR TITLE
Fix RT kernel DTK setting

### PR DESCRIPTION
**Pull request by pacevedom**
When using RT kernels the DTK image is not set, rendering broken BuildConfigs. This was due to a bug when retrieving cluster info after matching nodes and DTK metadata.

### CODE REVIEW

**Comment by yevgeny-shnaidman** [comment]
the code looks good.
can you also add a unit-test for this function? it will help us catch future bug

**Comment by pacevedom** [comment]
@yevgeny-shnaidman was planning to add tests as another PR, but why not :)
Thanks.

File: pkg/upgrade/upgrade_test.go [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+				},
+			},
+			dtkInput: registry.DriverToolkitEntry{
+				KernelFullVersion:   "4.18.0-305.19.1.el8_4",
+				RTKernelFullVersion: "4.18.0-305.19.1.rt7.91.el8_4",
+				OSVersion:           "8.4",
+			},
+			imageURLInput: "quay.io/somerepo/someimage@sha256:1234567890abcdef",
+			infoOutput:    nil,
+			errorOutput:   fmt.Errorf("OSVersion mismatch NFD: %s vs. DTK: %s", "8.2", "8.4"),
```
Comment (qbarrand) [inline pkg/upgrade/upgrade_test.go]:
```suggestion
			errorOutput:   errors.New("OSVersion mismatch NFD: 8.2 vs. DTK: 8.4"),
```

File: pkg/upgrade/upgrade_test.go [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+				},
+			},
+			dtkInput: registry.DriverToolkitEntry{
+				KernelFullVersion:   "4.18.0-305.19.1.el8_4",
+				RTKernelFullVersion: "4.18.0-305.19.1.rt7.91.el8_4",
+				OSVersion:           "8.4",
+			},
+			imageURLInput: "quay.io/somerepo/someimage@sha256:1234567890abcdef",
+			infoOutput:    nil,
+			errorOutput:   fmt.Errorf("OSVersion mismatch NFD: %s vs. DTK: %s", "8.1", "8.4"),
```
Comment (qbarrand) [inline pkg/upgrade/upgrade_test.go]:
```suggestion
			errorOutput:   errors.New("OSVersion mismatch NFD: 8.1 vs. DTK: 8.4"),
```

**Review by qbarrand — COMMENTED** [review]
Can the test errors be created with `errors.New`? Otherwise, LGTM.

**Pull request merged**